backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BundleRepository.java

```diff
@@ -82,4 +82,27 @@ public List<Application> getApplications(UUID id) {
                     .getResultList();
         }
     }
+
+    /**
+     * Finds the bundle by name.
+     * @param bundleName the name to find the bundle by.
+     * @return the found bundle from the database.
+     */
+    public Bundle findByName(final String bundleName) {
+        final String findByNameQuery =
+            "SELECT " +
+                    "b " +
+            "FROM " +
+                "Bundle AS b " +
+            "WHERE " +
+                "b.name = :name";
+
+        try {
+            return this.entityManager.createQuery(findByNameQuery, Bundle.class)
+                .setParameter("name", bundleName)
+                .getSingleResult();
+        } catch (final NoResultException e) {
+            throw new NotFoundException("the provided bundle name does not exist");
```

Comment:
Nitpick:

Throwing a JAX-RS exception here makes this method tightly-coupled with the REST layer while we could have a cleaner separation between the persistence and the REST layers.

This method could return an `Optional<Bundle>` and `Optional.empty()` in case of `NoResultException` to avoid that tight-coupling between layers. Then the calling method would throw the `NotFoundException` when the optional is empty.

Reply:
Yeah, makes a lot of sense. Dealt with!